src/devices/Ads1115/Ads1115.cs

```diff
@@ -14,12 +16,86 @@ namespace Iot.Device.Ads1115
     /// </summary>
     public class Ads1115 : IDisposable
     {
+        private readonly bool _shouldDispose;
+
         private I2cDevice _i2cDevice = null;
 
+        private GpioController _gpioController;
+
         private InputMultiplexer _inputMultiplexer;
 
+        private MeasuringRange _measuringRange;
+
+        private DataRate _dataRate;
+        private DeviceMode _deviceMode;
+
+        /// <summary>
+        /// The pin of the GPIO controller that is connected to the interrupt line of the ADS1115
+        /// </summary>
+        private int _gpioInterruptPin;
+
+        private ComparatorPolarity _comparatorPolarity;
+        private ComparatorLatching _comparatorLatching;
+        private ComparatorQueue _comparatorQueue;
+
+        /// <summary>
+        /// Initialize a new Ads1115 device connected through I2C
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="inputMultiplexer">Input Multiplexer</param>
+        /// <param name="measuringRange">Programmable Gain Amplifier</param>
+        /// <param name="dataRate">Data Rate</param>
+        /// <param name="deviceMode">Initial device mode</param>
+        public Ads1115(I2cDevice i2cDevice, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096, DataRate dataRate = DataRate.SPS128,
+            DeviceMode deviceMode = DeviceMode.Continuous)
+        {
+            _i2cDevice = i2cDevice;
```

Comment:
I know we weren't doing it before either, but should we be checking for null here in case i2cDevice passed it was null?